### DISCUSSION TITLE
Can cheatcodes be invoked outside of solidity tests/scripts?

### DISCUSSION THREAD

**Discussion by 0xIsshiki** [foundry-rs/foundry#3640] [Help]
I'd like to call cheatcodes to an `anvil` instance on a forked mainnet from web3js (in an integration test) to set up initial state. I assumed you could achieve this by loading the contract at `0x7109709ECfa91a80626fF3989D68f67F5b1DD12D` and calling a cheatcode function. But this does not seem to work for me (calling `roll` did not change the block number in subsequent calls). Am I doing something wrong or is this a limitation of cheatcodes?

**Comment by Vitalsine85** [comment]
Just came looking for an answer to this myself. I'll be following along. If I figure it out in the meantime, I'll be sure to share.

**Comment by mds1** [comment]
This is not currently supported, since the cheatcodes operate on the forge VM in-process, but anvil is a different process. Most likely the way to support this will be via a `vm.rpc` cheat so you can send the necessary RPC calls to anvil from a script more easily. This is being tracked in https://github.com/foundry-rs/foundry/issues/3362
